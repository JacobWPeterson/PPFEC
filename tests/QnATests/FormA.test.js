import React from 'react';
import { mount } from 'enzyme';
import FormA from '../../client/src/components/QnAComponents/FormAnswer.jsx';

describe('This is for the Answer Form', () => {
  const props = {
    id: 1234,
    name: 'Big Rock',
    body: 'asdf',
  };

  const FormAComp = mount(<FormA {...props} />);

  it('Should render Answer Form properly', () => {
    expect(FormAComp).toMatchSnapshot();
  });
});
