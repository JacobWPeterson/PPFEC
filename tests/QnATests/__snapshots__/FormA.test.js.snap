// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is for the Answer Form Should render Answer Form properly 1`] = `
<FormA
  body="asdf"
  id={1234}
  name="Big Rock"
>
  <FormAnswer__FormAContainer>
    <form
      className="FormAnswer__FormAContainer-xot12g-0 dmqVLc"
    >
      SUBMIT YOUR ANSWER
      <h4>
        BIG ROCK
        : 
        ASDF
      </h4>
      <FormAnswer__Pair>
        <div
          className="FormAnswer__Pair-xot12g-6 ndBCE"
        >
          <FormAnswer__Title>
            <h4
              className="FormAnswer__Title-xot12g-7"
            >
              HOW CAN YOU HELP?
            </h4>
          </FormAnswer__Title>
          <FormAnswer__BodyT
            cols="55"
            maxLength="1000"
            name="body"
            onChange={[Function]}
            placeholder="HELP SOMEONE OUT HERE"
            required={true}
            rows="5"
            type="text"
            value=""
          >
            <textarea
              className="FormAnswer__BodyT-xot12g-2 eWEeUV"
              cols="55"
              maxLength="1000"
              name="body"
              onChange={[Function]}
              placeholder="HELP SOMEONE OUT HERE"
              required={true}
              rows="5"
              type="text"
              value=""
            />
          </FormAnswer__BodyT>
        </div>
      </FormAnswer__Pair>
      <FormAnswer__Pair>
        <div
          className="FormAnswer__Pair-xot12g-6 ndBCE"
        >
          <FormAnswer__Title>
            <h4
              className="FormAnswer__Title-xot12g-7"
            >
              MAY I HAVE YOUR NAME?
            </h4>
          </FormAnswer__Title>
          <FormAnswer__Input
            name="name"
            onChange={[Function]}
            placeholder="ADD YOUR NAME HERE"
            value=""
          >
            <input
              className="FormAnswer__Input-xot12g-1 jfpGjI"
              name="name"
              onChange={[Function]}
              placeholder="ADD YOUR NAME HERE"
              value=""
            />
          </FormAnswer__Input>
          <FormAnswer__BText>
            <div
              className="FormAnswer__BText-xot12g-11 cyKKsS"
            >
              FOR PRIVACY REASONS, DO NOT USE YOUR FULL NAME OR EMAIL ADDRESS
            </div>
          </FormAnswer__BText>
        </div>
      </FormAnswer__Pair>
      <FormAnswer__Pair>
        <div
          className="FormAnswer__Pair-xot12g-6 ndBCE"
        >
          <FormAnswer__Title>
            <h4
              className="FormAnswer__Title-xot12g-7"
            >
              HOW ABOUT YOUR EMAIL?
            </h4>
          </FormAnswer__Title>
          <FormAnswer__Input
            name="email"
            onChange={[Function]}
            placeholder="EXAMPLE: jack@email.com"
            value=""
          >
            <input
              className="FormAnswer__Input-xot12g-1 jfpGjI"
              name="email"
              onChange={[Function]}
              placeholder="EXAMPLE: jack@email.com"
              value=""
            />
          </FormAnswer__Input>
          <FormAnswer__BText>
            <div
              className="FormAnswer__BText-xot12g-11 cyKKsS"
            >
              FOR AUTHENTICATION REASONS, YOU WILL NOT BE EMAILED
            </div>
          </FormAnswer__BText>
        </div>
      </FormAnswer__Pair>
      <FormAnswer__Pair>
        <div
          className="FormAnswer__Pair-xot12g-6 ndBCE"
        >
          <FormAnswer__Title>
            <h4
              className="FormAnswer__Title-xot12g-7"
            >
              ADD A PHOTO
            </h4>
          </FormAnswer__Title>
          <FormAnswer__PhotoC>
            <div
              className="FormAnswer__PhotoC-xot12g-9 kKzexq"
            >
              <FormAnswer__PhotoI
                name="photos"
                onChange={[Function]}
                onSubmit={[Function]}
                placeholder="ADD URL"
                value=""
              >
                <input
                  className="FormAnswer__PhotoI-xot12g-10 jdjuOu"
                  name="photos"
                  onChange={[Function]}
                  onSubmit={[Function]}
                  placeholder="ADD URL"
                  value=""
                />
              </FormAnswer__PhotoI>
              <FormAnswer__AddP
                onClick={[Function]}
              >
                <button
                  className="FormAnswer__AddP-xot12g-8 eqUxmR"
                  onClick={[Function]}
                >
                  ADD PHOTO
                </button>
              </FormAnswer__AddP>
            </div>
          </FormAnswer__PhotoC>
          <FormAnswer__BText>
            <div
              className="FormAnswer__BText-xot12g-11 cyKKsS"
            >
              YOU CAN ADD UP TO FIVE, YOU'VE ADDED (0) SO FAR
            </div>
          </FormAnswer__BText>
        </div>
      </FormAnswer__Pair>
      <FormAnswer__Buttons>
        <div
          className="FormAnswer__Buttons-xot12g-3 jXzctE"
        >
          <FormAnswer__BP>
            <div
              className="FormAnswer__BP-xot12g-5 hEyJIA"
            >
              <FormAnswer__BC
                name="Submit"
                onClick={[Function]}
              >
                <button
                  className="FormAnswer__BC-xot12g-4 dlwryK"
                  name="Submit"
                  onClick={[Function]}
                >
                  SUBMIT
                </button>
              </FormAnswer__BC>
            </div>
          </FormAnswer__BP>
          <FormAnswer__BP>
            <div
              className="FormAnswer__BP-xot12g-5 hEyJIA"
            >
              <FormAnswer__BC
                name="Exit"
                onClick={[Function]}
              >
                <button
                  className="FormAnswer__BC-xot12g-4 dlwryK"
                  name="Exit"
                  onClick={[Function]}
                >
                  EXIT
                </button>
              </FormAnswer__BC>
            </div>
          </FormAnswer__BP>
        </div>
      </FormAnswer__Buttons>
    </form>
  </FormAnswer__FormAContainer>
</FormA>
`;

exports[`This is for the Answer Form Should render properly 1`] = `
<FormA
  body="asdf"
  id={1234}
  name="Big Rock"
>
  <FormAnswer__FormAContainer>
    <form
      className="FormAnswer__FormAContainer-xot12g-0 dmqVLc"
    >
      SUBMIT YOUR ANSWER
      <h4>
        BIG ROCK
        : 
        ASDF
      </h4>
      <FormAnswer__Pair>
        <div
          className="FormAnswer__Pair-xot12g-6 ndBCE"
        >
          <FormAnswer__Title>
            <h4
              className="FormAnswer__Title-xot12g-7"
            >
              HOW CAN YOU HELP?
            </h4>
          </FormAnswer__Title>
          <FormAnswer__BodyT
            cols="55"
            maxLength="1000"
            name="body"
            onChange={[Function]}
            placeholder="HELP SOMEONE OUT HERE"
            required={true}
            rows="5"
            type="text"
            value=""
          >
            <textarea
              className="FormAnswer__BodyT-xot12g-2 eWEeUV"
              cols="55"
              maxLength="1000"
              name="body"
              onChange={[Function]}
              placeholder="HELP SOMEONE OUT HERE"
              required={true}
              rows="5"
              type="text"
              value=""
            />
          </FormAnswer__BodyT>
        </div>
      </FormAnswer__Pair>
      <FormAnswer__Pair>
        <div
          className="FormAnswer__Pair-xot12g-6 ndBCE"
        >
          <FormAnswer__Title>
            <h4
              className="FormAnswer__Title-xot12g-7"
            >
              MAY I HAVE YOUR NAME?
            </h4>
          </FormAnswer__Title>
          <FormAnswer__Input
            name="name"
            onChange={[Function]}
            placeholder="ADD YOUR NAME HERE"
            value=""
          >
            <input
              className="FormAnswer__Input-xot12g-1 jfpGjI"
              name="name"
              onChange={[Function]}
              placeholder="ADD YOUR NAME HERE"
              value=""
            />
          </FormAnswer__Input>
          <FormAnswer__BText>
            <div
              className="FormAnswer__BText-xot12g-11 cyKKsS"
            >
              FOR PRIVACY REASONS, DO NOT USE YOUR FULL NAME OR EMAIL ADDRESS
            </div>
          </FormAnswer__BText>
        </div>
      </FormAnswer__Pair>
      <FormAnswer__Pair>
        <div
          className="FormAnswer__Pair-xot12g-6 ndBCE"
        >
          <FormAnswer__Title>
            <h4
              className="FormAnswer__Title-xot12g-7"
            >
              HOW ABOUT YOUR EMAIL?
            </h4>
          </FormAnswer__Title>
          <FormAnswer__Input
            name="email"
            onChange={[Function]}
            placeholder="EXAMPLE: jack@email.com"
            value=""
          >
            <input
              className="FormAnswer__Input-xot12g-1 jfpGjI"
              name="email"
              onChange={[Function]}
              placeholder="EXAMPLE: jack@email.com"
              value=""
            />
          </FormAnswer__Input>
          <FormAnswer__BText>
            <div
              className="FormAnswer__BText-xot12g-11 cyKKsS"
            >
              FOR AUTHENTICATION REASONS, YOU WILL NOT BE EMAILED
            </div>
          </FormAnswer__BText>
        </div>
      </FormAnswer__Pair>
      <FormAnswer__Pair>
        <div
          className="FormAnswer__Pair-xot12g-6 ndBCE"
        >
          <FormAnswer__Title>
            <h4
              className="FormAnswer__Title-xot12g-7"
            >
              ADD A PHOTO
            </h4>
          </FormAnswer__Title>
          <FormAnswer__PhotoC>
            <div
              className="FormAnswer__PhotoC-xot12g-9 kKzexq"
            >
              <FormAnswer__PhotoI
                name="photos"
                onChange={[Function]}
                onSubmit={[Function]}
                placeholder="ADD URL"
                value=""
              >
                <input
                  className="FormAnswer__PhotoI-xot12g-10 jdjuOu"
                  name="photos"
                  onChange={[Function]}
                  onSubmit={[Function]}
                  placeholder="ADD URL"
                  value=""
                />
              </FormAnswer__PhotoI>
              <FormAnswer__AddP
                onClick={[Function]}
              >
                <button
                  className="FormAnswer__AddP-xot12g-8 eqUxmR"
                  onClick={[Function]}
                >
                  ADD PHOTO
                </button>
              </FormAnswer__AddP>
            </div>
          </FormAnswer__PhotoC>
          <FormAnswer__BText>
            <div
              className="FormAnswer__BText-xot12g-11 cyKKsS"
            >
              YOU CAN ADD UP TO FIVE, YOU'VE ADDED (0) SO FAR
            </div>
          </FormAnswer__BText>
        </div>
      </FormAnswer__Pair>
      <FormAnswer__Buttons>
        <div
          className="FormAnswer__Buttons-xot12g-3 jXzctE"
        >
          <FormAnswer__BP>
            <div
              className="FormAnswer__BP-xot12g-5 hEyJIA"
            >
              <FormAnswer__BC
                name="Submit"
                onClick={[Function]}
              >
                <button
                  className="FormAnswer__BC-xot12g-4 dlwryK"
                  name="Submit"
                  onClick={[Function]}
                >
                  SUBMIT
                </button>
              </FormAnswer__BC>
            </div>
          </FormAnswer__BP>
          <FormAnswer__BP>
            <div
              className="FormAnswer__BP-xot12g-5 hEyJIA"
            >
              <FormAnswer__BC
                name="Exit"
                onClick={[Function]}
              >
                <button
                  className="FormAnswer__BC-xot12g-4 dlwryK"
                  name="Exit"
                  onClick={[Function]}
                >
                  EXIT
                </button>
              </FormAnswer__BC>
            </div>
          </FormAnswer__BP>
        </div>
      </FormAnswer__Buttons>
    </form>
  </FormAnswer__FormAContainer>
</FormA>
`;
